src/pages/index.js

```diff
@@ -0,0 +1,328 @@
+    .then((data) => {
+      avatarImage.src = data.avatar;
+      avatarInput.value = "";
+      disabledButton(avatarSubmitButton, settings);
+      hideAllModals();
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitbtn.textContent = "Save";
+    });
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__caption");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLike = cardElement.querySelector(".card__icon");
+  const cardDelete = cardElement.querySelector(".card__delete-icon");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardLike.addEventListener("click", (evt) => {
+    handleLike(evt, data._id, cardLike);
+  });
+  cardImageEl.addEventListener("click", () => {
+    openPopup(previewModal);
+    previewModalCaption.textContent = data.name;
+    previewModalImage.src = data.link;
+    previewModalImage.alt = data.name;
+  });
+
+  cardDelete.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  return cardElement;
+}
+
+function handleLike(evt, id, cardLikeElement) {
+  const isLiked = cardLikeElement.classList.contains("card__icon_liked");
+  api
+    .updateLikeStatus(id, isLiked)
+    .then((data) => {
+      if (isLiked) {
+        cardLikeElement.classList.remove("card__icon_liked");
+      } else {
+        cardLikeElement.classList.add("card__icon_liked");
+      }
+    })
+    .catch(console.error);
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openPopup(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitbtn = evt.submitter;
+  submitbtn.textContent = "Deleting...";
+  api
+    .removeCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closePopup(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitbtn.textContent = "Delete";
+    });
+}
+function closePopup(popup) {
+  popup.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscape);
+}
+
+function openPopup(popup) {
+  popup.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscape);
+}
+function showProfileModal() {
+  openPopup(editModal);
+  editNameInput.value = profileNameElement.textContent;
+  editJobInput.value = profileJobElement.textContent;
+}
+
+function hideAllModals() {
+  [editModal, postModal, previewModal, deleteModal, avatarModal].forEach(
+    closePopup
+  );
+}
+
+[
+  editModalCloseButton,
+  postModalCloseButton,
+  previewCloseButton,
+  deleteCloseButton,
+  deleteCancelButton,
+  deleteCloseImage,
+  avatarModalCloseButton,
+].forEach((button) => {
+  button.addEventListener("click", hideAllModals);
```

Comment:
**Could be improved**

You can make a universal handler for any close buttons.

It will look something like this:

```js
// Find all close buttons
const closeButtons = document.querySelectorAll('.modal__close');

closeButtons.forEach((button) => {
  // Find the closest popup only once
  const popup = button.closest('.modal');
  // Set the listener
  button.addEventListener('click', () => closePopup(popup));
});



```

That’s why we make universal css classes for similar elements: here it’s `modal__close` for any close button.

So now you can add 100 modals: their close buttons will be handled automatically.